controllers/clustermanagementaddon_controller.go

```diff
@@ -152,43 +153,52 @@ func (c *ClusterManagementAddonReconciler) Reconcile(ctx context.Context, reques
 	}
 
 	// deploying central proxy server instances into the hub cluster.
-	err = c.deployProxyServer(config)
+	isRedeployed, err := c.deployProxyServer(config)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
+	if !isRedeployed {
+		klog.Infof("Proxy server resources are up-to-date")
+	}
 
 	// refreshing status
-	if err := c.refreshStatus(config); err != nil {
+	if err := c.refreshStatus(isRedeployed, config); err != nil {
 		return reconcile.Result{}, err
 	}
 	return reconcile.Result{}, nil
 }
 
-func (c *ClusterManagementAddonReconciler) refreshStatus(config *proxyv1alpha1.ManagedProxyConfiguration) error {
-	currentState, err := c.getCurrentState(config)
+func (c *ClusterManagementAddonReconciler) refreshStatus(isRedeployed bool, config *proxyv1alpha1.ManagedProxyConfiguration) error {
+	currentState, err := c.getCurrentState(isRedeployed, config)
 	if err != nil {
 		return err
 	}
-	status := proxyv1alpha1.ManagedProxyConfigurationStatus{}
-	status.LastObservedGeneration = config.Generation
-	status.Conditions = c.getConditions(currentState)
-	mungedStatus := config.Status.DeepCopy()
-	for i := range mungedStatus.Conditions {
-		mungedStatus.Conditions[i].LastTransitionTime = metav1.Time{}
+	expectingStatus := proxyv1alpha1.ManagedProxyConfigurationStatus{}
+	expectingStatus.LastObservedGeneration = config.Generation
+	expectingStatus.Conditions = c.getConditions(currentState)
+	currentStatus := config.Status.DeepCopy()
+	for i := range currentStatus.Conditions {
+		currentStatus.Conditions[i].LastTransitionTime = metav1.Time{}
 	}
-	if equality.Semantic.DeepEqual(&status, mungedStatus) {
+	if !isRedeployed && equality.Semantic.DeepEqual(&expectingStatus, currentStatus) {
 		return nil
 	}
 	now := metav1.Now()
 	editingConfig := config.DeepCopy()
-	editingConfig.Status = status
+	editingConfig.Status = expectingStatus
 	for i := range editingConfig.Status.Conditions {
-		editingConfig.Status.Conditions[i].LastTransitionTime = now
+		expectingCondition := editingConfig.Status.Conditions[i]
+		currentCondition := meta.FindStatusCondition(config.Status.Conditions, expectingCondition.Type)
```

Comment:
I think you can use meta.SetStatusCondition here which will handle transition time.

Reply:
updated